dataset/dataset.go

```diff
@@ -1018,6 +1018,43 @@ func (c *Client) PostInstanceDimensions(ctx context.Context, serviceAuthToken, i
 	return eTag, nil
 }
 
+// PatchInstanceDimensions performs a 'PATCH /instances/<id>/dimensions' with the provided List of Options to patch
+func (c *Client) PatchInstanceDimensions(ctx context.Context, serviceAuthToken, instanceID string, data []*OptionPost, ifMatch string) (eTag string, err error) {
+	uri := fmt.Sprintf("%s/instances/%s/dimensions", c.hcCli.URL, instanceID)
+
+	if len(data) == 0 {
+		log.Info(ctx, "skipping patch call because no update was provided", log.Data{"uri": uri})
+		return ifMatch, nil
+	}
+
+	patchBody := []dprequest.Patch{
+		{
+			Op:    dprequest.OpAdd.String(),
```

Comment:
on end of this line, please add comment:
// this will cause an 'upsert' to be actioned